chia/rpc/wallet_rpc_api.py

```diff
@@ -1383,10 +1384,25 @@ async def dl_latest_singleton(self, request) -> Dict:
 
         raise ValueError("No DataLayer wallet has been initialized")
 
+    async def dl_singletons_by_root(self, request) -> Dict:
+        """Get the singleton records that contain the specified root"""
+        if self.service.wallet_state_manager is None:
+            raise ValueError("The wallet service is not currently initialized")
+
+        for wallet in self.service.wallet_state_manager.wallets.values():
```

Comment:
i would do a get_data_wallet method so we can avoid for looping any time we need the wallet

Reply:
I have a local branch for this and other changes that I'll PR when general changes will break less other PRs.